script/release.js

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+
+const { versionBump, currentVersion }  = require('./version-utils');
+const { logger, spawnOrFail, prompt, shouldContinuePrompt, quit, process, path } = require('./cli-utils');
+
+const deployDemo = (version) => {
+  const demo_name = `chime-sdk-demo-${version.replace(/\./g, "-")}`;
+  logger.log(`Deploying ${demo_name} ...`);
+  process.chdir(path.join(__dirname, '../demos/serverless'));
+  spawnOrFail('npm', [`run deploy -- -b ${demo_name} -s ${demo_name} -o ${demo_name} -u false`], { printErr: true });
+};
+
+const getCurrentRemoteBranch = () => {
+  return (spawnOrFail('git', ['for-each-ref --format="%(upstream:short)" "$(git symbolic-ref -q HEAD)"'], { skipOutput: true })).trim();
+};
+
+const buildAndPack = () => {
+  logger.log('Building package...');
+  spawnOrFail('npm', ['run build:release']);
+  logger.log('Packaging ...');
+  spawnOrFail('npm', ['pack --dry-run'], { printErr: true });
+};
+
+const cleanUp = async (remoteBranch) => {
+  logger.warn(`Warning: Resetting HEAD ${remoteBranch ? `to ${remoteBranch}` : ''}.\nAll current staged and local changes will be lost.`);
+  await shouldContinuePrompt();
+  spawnOrFail('git', [`reset --hard ${remoteBranch ? remoteBranch : ''}`]);
+  spawnOrFail('git', [' clean -ffxd .']);
+};
+
+const release = async () => {
+  spawnOrFail('git', ['fetch origin'], { skipOutput: true });
+  const currentBranch = (spawnOrFail('git', [' branch --show-current'], { skipOutput: true })).trim();
+  const remoteBranch = getCurrentRemoteBranch();
+  if (!remoteBranch || (remoteBranch !== 'origin/main' && (/^origin\/release-[0-9]+\.x$/).test(remoteBranch))) {
+    logger.error(`The local branch ${currentBranch} does not track either main or release-<version>.x branch`);
+    quit(1);
+  }
+  await cleanUp(remoteBranch);
+
+  buildAndPack();
+  logger.log('Do you want to upload these files to release branch?\n');
+  await shouldContinuePrompt();
+  spawnOrFail('git', ['push origin HEAD:release -f']);
+  deployDemo(currentVersion);
+
+  //Bump next development version
+  await versionBump();
+};
+
+const hotfix = async () => {
+  await cleanUp();
+
+  buildAndPack();
+
+  await versionBump(1, 'hotfix');
```

Comment:
This will result into `undefined` for a pre-release version so what will be doing for hotfix for a pre-release version?

Reply:
I don't think we hotfix a pre-release version right since it is beta and expected to have bugs :D Guess something we need to consider :))

Reply:
So should we then drop this method call?

Reply:
what do you mean dropping it? We have to increase the patch version for hotfix right? Or do you mean to have some error handling if the current version is a pre-release version?

Reply:
Yes right, I meant just check here itself to not do anything if a pre-release. Needs some implicit understanding that we wont do any hotfix for a pre-release testing version. I think a comment can also suffice.

Reply:
Added error out for pre-release version.